Source/Core/getStringFromTypedArray.js

```diff
@@ -0,0 +1,49 @@
+/*global define*/
+define([
+        './defined',
+        './DeveloperError'
+    ], function(
+        defined,
+        DeveloperError) {
+    "use strict";
+    /*global TextDecoder*/
+
+    /**
+     * @private
+     */
+    var getStringFromTypedArray = function(buffer, byteOffset, length) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(buffer)) {
+            throw new DeveloperError('buffer is required.');
+        }
+
+        if (!defined(byteOffset)) {
+            throw new DeveloperError('byteOffset is required.');
+        }
+
+        if (!defined(length)) {
+            throw new DeveloperError('length is required.');
+        }
+        //>>includeEnd('debug');
+
+        return getStringFromTypedArray.decode(new Uint8Array(buffer, byteOffset, length));
+    };
+
+    // Exposed functions for testing
+    getStringFromTypedArray.decodeWithTextDecoder = function(view) {
+        var decoder = new TextDecoder('utf-8');
+        return decoder.decode(view);
+    };
+
+    getStringFromTypedArray.decodeWithFromCharCode = function(view) {
+        return String.fromCharCode.apply(String, view);
```

Comment:
> fromCharCode is not going to work in all cases

I added a comment.  Note that this is a fallback for `TextDecoder`.

@jbo023 thanks for the workaround.  It's not pretty but we'll go with it since this is a fallback anyway.